From: Andrew Cann <shum@canndrew.org>
Description: Fix build error due to failed type inference
diff -Nru fsharp-4.5.0.orig/src/scripts/fssrgen.fsx fsharp-4.5.0/src/scripts/fssrgen.fsx
index 32c5b4fa5..ffd155747 100644
--- fsharp-4.5.0.orig/src/scripts/fssrgen.fsx
+++ fsharp-4.5.0/src/scripts/fssrgen.fsx
@@ -332,7 +332,7 @@ let StringBoilerPlate filename =
     // END BOILERPLATE        
 "            
 
-let RunMain(filename, outFilename, outXmlFilenameOpt, projectNameOpt) =
+let RunMain(filename: string, outFilename, outXmlFilenameOpt, projectNameOpt) =
     try
         let justfilename = System.IO.Path.GetFileNameWithoutExtension(filename)
         if justfilename |> Seq.exists (fun c -> not(System.Char.IsLetterOrDigit(c))) then
diff -Nru fsharp-4.5.0.orig/src/scripts/scriptlib.fsx fsharp-4.5.0/src/scripts/scriptlib.fsx
index 2059844e0..daf8d32fa 100644
--- fsharp-4.5.0.orig/src/scripts/scriptlib.fsx
+++ fsharp-4.5.0/src/scripts/scriptlib.fsx
@@ -93,7 +93,7 @@ module Scripting =
 
     module Process =
 
-        let processExePath baseDir exe =
+        let processExePath baseDir (exe: FilePath) =
             if Path.IsPathRooted(exe) then exe
             else 
                 match Path.GetDirectoryName(exe) with
