Index: fsharp/src/FSharpSource.targets
===================================================================
--- fsharp.orig/src/FSharpSource.targets
+++ fsharp/src/FSharpSource.targets
@@ -983,8 +983,4 @@ Some other NuGET monikers to support in
   <PropertyGroup>
     <SolutionDir>$(FSharpSourcesRoot)\..\</SolutionDir>
   </PropertyGroup>
-  <Import Project="$(FSharpSourcesRoot)\..\.nuget\NuGet.targets" />
-  <Target Name="BeforeBuild" BeforeTargets="Build">
-    <Exec Command="$(NuGetCommand) restore packages.config -PackagesDirectory packages" WorkingDirectory="$(FSharpSourcesRoot)\.."/>
-  </Target>
 </Project>
